src/fidesops/main.py

```diff
@@ -40,6 +42,75 @@
         allow_headers=["*"],
     )
 
+
+@app.middleware("http")
+async def dispatch_log_request(request: Request, call_next: Callable) -> Response:
```

Comment:
It seems like this middleware could be unit-tested, you could assert that certain calls are made.  That brings up a second question, how are you preventing analytics from being sent with unit tests?

Reply:
Good call, I'll add some tests here. 

To answer your question, we add the `ANALYTICS_OPT_OUT` flag within `Makefile` when we run unit/integration tests or linters. This means that analytics will still be sent when bypassing Makefile and calling `pytest` directly from `make server` or pyenv. But, at least internal folks will have the `INTERNAL` client id to help with this.

Reply:
Could you add an autouse fixture to `conftest.py` to prevent it from sending? I haven't looked through yet to see if there is a good patch point for that so the answer maybe no :smile:

Reply:
testing out something like this:

```
@pytest.fixture(autouse=True, scope="session")
def analytics_opt_out():
    """Disable sending analytics when running tests."""
    original_value = os.getenv("FIDESOPS__ROOT_USER__ANALYTICS_OPT_OUT")
    os.environ["FIDESOPS__ROOT_USER__ANALYTICS_OPT_OUT"] = "true"
    yield
    os.environ["FIDESOPS__ROOT_USER__ANALYTICS_OPT_OUT"] = original_value
```

Reply:
Setting this os var isn't working as written above ^ and not sure why yet

Reply:
looks like setting config worked:

```
@pytest.fixture(autouse=True, scope="session")
def analytics_opt_out():
    """Disable sending analytics when running tests."""
    original_value = config.root_user.ANALYTICS_OPT_OUT
    config.root_user.ANALYTICS_OPT_OUT = True
    yield
    config.root_user.ANALYTICS_OPT_OUT = original_value
```

Reply:
Looks like a good solution to me

Reply:
fixture looks good @eastandwestwind 👍